examples/kitchen-sink/kitchen-sink-schema.graphql

```diff
@@ -51,16 +70,151 @@ input NestedType {
   veryNested: String
 }
 
+"""
+PageInfo cursor, as defined in https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo
+"""
+type PageInfo {
+  """
+  The cursor corresponding to the last nodes in edges. Null if the connection is empty.
+  """
+  endCursor: String
```

Comment:
The [spec](https://facebook.github.io/relay/graphql/connections.htm#sec-undefined.PageInfo) says `startCursor` and `endCursor` are non-null:

> It must also contain fields startCursor and endCursor, both of which return non‐null opaque strings.

I think for this to be spec-compliant, the entire connection field on an object type needs to be nullable to account for a connection not having any edges to return.

Reply:
> The spec says startCursor and endCursor are non-null:

I did see that, though I believe this is either incorrect or outdated. 

For instance, in graphql-relay-js, the official implementation, start and end cursor are non-null:

https://github.com/graphql/graphql-relay-js/blob/84ed0008ee17ae439ce67d820a29d1f258addff8/src/connection/connection.js#L147-L154

There is also this commit on the official relay project which demonstrates that the cursors are null when there are no results:

https://github.com/facebook/relay/commit/a17b462b3ff7355df4858a42ddda75f58c161302

There is an open PR addressing this inconsistency, but it looks like it needs an update by the person opening the PR:

https://github.com/facebook/relay/pull/2655

Thanks for taking the time to take a look though @swac, let me know if there's anything else you see in the PR that could use some improvements, I'm planning on cutting a `rc` release for it later today.